### PR TITLE
Adding food command for foodspotting

### LINKED ISSUE #151: Make marty send food spotting
Users can call `?food` to add food to the foodspotting channel
(Discussed in #idoneam)

### PR DESCRIPTION
**Pull request by le-potate**
Resolves #151 
This adds a food command that posts messages to the food spotting channel. Also, channels are now all taken by name in config.ini

### CODE REVIEW

File: cogs/helpers.py [hunk 1]
```diff
@@ -510,6 +510,31 @@ def __init__(self, bot):
         except ValueError:
             ctx.send("Input must be integers")
 
+    @commands.command()
+    async def food(self, ctx, *args):
```
Comment (jidicula) [inline cogs/helpers.py]: Can you change the command name to something more related to its function, for example ?foodspot

File: cogs/helpers.py [hunk 2]
```diff
@@ -510,6 +510,31 @@ def __init__(self, bot):
...
+        """Posts a food sale in #foodspotting.
+        Use: `?food Samosas in leacock`
+        You can also attach one picture to your message"""
```
Comment (jidicula) [inline cogs/helpers.py]: Does the command have to be added in the uploaded image caption or can the image be added in the next message after the command?

File: cogs/helpers.py [hunk 3]
```diff
@@ -510,6 +510,31 @@ def __init__(self, bot):
...
+        channel = utils.get(self.bot.get_guild(
+            self.bot.config.server_id).text_channels,
+                            name=self.bot.config.food_spotting_channel)
+        username = ctx.message.author
+        pfp = ctx.message.author.avatar_url
+        embed = discord.Embed()
+        try:
+            embed.set_image(url=ctx.message.attachments[0].url)
+        except:
+            pass
```
Comment (jidicula) [inline cogs/helpers.py]: thank you for not leaving a bare `except`

File: cogs/mod.py [hunk 4]
```diff
@@ -49,9 +51,10 @@ def __init__(self, bot):
         dest = user
         await dest.send(
             content='{}\n*To answer write* `{}answer "your message here"`'.
-            format(message, self.bot.config.command_prefix))
-        channel_to_forward = self.bot.get_channel(
-            self.bot.config.reception_channel_id)
+            format(message, self.bot.config.command_prefix[0]))
```
Comment (davidlougheed) [inline cogs/mod.py]: ok as an aside why is this an array in the first place? can there be more than one?

File: cogs/mod.py [hunk 5]
```diff
@@ -34,8 +35,9 @@ def __init__(self, bot):
             message = "{}".format(
                 " ".join(args)
             )    # to work regardless of whether the person uses apostrophes
-            channel_to_send = self.bot.get_channel(
-                self.bot.config.reception_channel_id)
+            channel_to_send = utils.get(self.bot.get_guild(
```
Comment (jidicula) [inline cogs/mod.py]: this should be in another issue branch and another commit. please follow [atomic commit](https://en.wikipedia.org/wiki/Atomic_commit) guidelines.

File: cogs/mod.py (hunk 4 above)
Comment (le-potate) [reply cogs/mod.py]: it seems so, but also when you send multiple pictures it always sends them in different messages so i'm not sure when this happens

File: config/config.ini [hunk 6]
```diff
@@ -28,7 +28,7 @@ UpvoteEmoji = upmartlet
 DownvoteEmoji = downmartlet
 ModeratorRole = Discord Moderator
 DeveloperRole = idoneam
-ReceptionChannelID = 454061583874785280
+ReceptionChannel = martys_dm
```
Comment (jidicula) [inline config/config.ini]: atomic commits smh

File: cogs/helpers.py (hunk 2 above)
Comment (jidicula) [reply cogs/helpers.py]: This should be specified in the help docstring.

Comment (le-potate) [reply cogs/helpers.py]: The command has to be added in the uploaded image caption

File: cogs/helpers.py (hunk 3 above)
Comment (jidicula) [reply cogs/helpers.py]: wait jk that *is* a bare except. you can instead put

```python
except Exception:
    pass:
```

File: config/config.ini (hunk 6 above)
Comment (le-potate) [reply config/config.ini]: will be careful next time

File: cogs/helpers.py (hunk 1 above)
Comment (jidicula) [reply cogs/helpers.py]: actually a shorter command name is better for mobile users

Comment (jidicula) [reply cogs/helpers.py]: e.g. ?fs

File: cogs/mod.py (hunk 4 above)
Comment (jidicula) [reply cogs/mod.py]: the iOS app allows multiple pics per message. not sure about android. definitely not an option on desktop.
